Kernel/interruptions/keyboard.c

```diff
@@ -83,12 +76,25 @@ void keyboard_handler() {
     }
 }
 
-int64_t getChar(void) {
-  if (head != tail) {
-    char ans = buffer[tail++];
-    if (tail == BUFFER_SIZE)
-      tail = 0;
-      return ans == 0 ? -1 : ans;
-  }
-  return -1;
+// int64_t getChar(void) {
+//   if (head != tail) {
+//     char ans = buffer[tail++];
+//     if (tail == BUFFER_SIZE)
+//       tail = 0;
+//       return ans == 0 ? -1 : ans;
+//   }
+//   return -1;
+// }
+
+long copy_from_buffer(char * buf, size_t count) {
+  if (r_pointer == w_pointer) return -1; // TODO: Check what we can do if the buffer "pega la vuelta".
+  
+  if (count > BUFFER_SIZE) {} // TODO: what do we do.
+  
+  long i = 0;
+
+  while (i < count && r_pointer != w_pointer)  // TODO: Check if we keep the last \n
+    buf[i++] = buffer[r_pointer++];
+  
+  return i;
 }
```

Comment:
Files should end on a new line.

Reply:
The clang-formatter should tajke care of this later